src/node/rpc/handler_registry.h

```diff
@@ -119,9 +189,24 @@ namespace ccf
         return *this;
       }
 
-      // If true, request is executed without consensus (PBFT only)
       bool execute_locally = false;
 
+      /** Indicates that the execution of the handler does not require
+       * consensus from other nodes in the network. This has effect only if PBFT
+       * is used as the underlying consensus algorithm.
```

Comment:
I don't think this _should_ be PBFT-only (not sure if it currently _is_). A node-status RPC should be local-only regardless of consensus choice. And I think the core detail is that it only _reads_ node-local state, not things which require consensus. Devs could provide as many of these as they like for per-node debugging, but in general they shouldn't need to. This also breaks our session consistency claims from elsewhere since it will never be forwarded (this is useful for some RPCs!). I think they shouldn't even read from the KV? But perhaps this is unavoidable.

Reply:
You're absolutely right. At the moment, the flag is only used to circumvent the broadcast and consensus on requests issued with PBFT, for the `getMetrics` and `getLocalCommit` endpoints. I agree that they shouldn't even read from the KV. I will had a note on that.